src/main/java/codeu/model/data/Event.java

```diff
@@ -43,6 +44,15 @@
   private final String nameOfUser;
   private final UUID userId;
 
+  /* All the fields needed from the media class. */
+  private final UUID mediaId;
+  private final UUID ownermediaId;
+  private final Instant mediaCreationTime;
+  private final String titleOfMedia;
+  private final BufferedImage contentOfMedia;
+  private final String contentTypeOfMedia;
+  private final UUID conversationIdForMedia;
```

Comment:
NOT NEED TO MAKE THIS CHANGE, but something to think about.

Instead of have all the fields for each EVENT_TYPE defined in Event class (and only a subset of these fields are used for each Event instance depending on the event type), a better design might be have Event be an abstract class, and sub-classes being, NewConversationEvent, NewUserEvent, MediaUploadEvent etc.

Reply:
That would be way better. I'll keep this in mind if I can keep working on this.

Reply:
https://www.quora.com/What-is-the-difference-between-inheritance-encapsulation-and-polymorphism
http://www.baeldung.com/java-inheritance-composition

A very common interview questions on these topics.